meson.build

```diff
@@ -0,0 +1,1449 @@
+if host_machine.system() == 'linux'
+   exportedVariables += {
+      'CONFIG_TARGET_INSTALL_DIR' : '/'
+   }
+endif
+
+#
+# Export the name/location of the desktop icon (which, on Windows, is also the same icon for the installer) for the
+# packaging scripts.
+#
+if host_machine.system() == 'windows'
+   exportedVariables += {
+      'CONFIG_INSTALLER_APPLICATION_ICON_PATH' : join_paths(meson.current_source_dir(), 'win', capitalisedProjectName + 'Icon_96.ico'),
+   }
+elif host_machine.system() == 'darwin'
+   exportedVariables += {
+      'CONFIG_INSTALLER_APPLICATION_ICON'      : capitalisedProjectName + 'Icon.icns',
+   }
+endif
+
+#=======================================================================================================================
+#===================================== Generate config.h etc from config.h.in etc ======================================
+#=======================================================================================================================
+
+#
+# First we inject build-system variables into the source code.  This saves defining certain things twice - in this file
+# and in a C++ source code file.  It also makes it easier for Brewken and Brewtarget to share code, because there are
+# fewer places where the application name is hard-coded.
+#
+# Taking src/config.h.in as input, we generate (in the build subdirectory only) config.h.
+#
+# All variables written as "@VAR@" in src/config.h.in (input file) will be replaced in config.h (output file) by the
+# value of VAR in the configuration_data dictionary we define here.
+#
+configure_file(input  : 'src/config.h.in',
+               output : 'config.h',
+               configuration : exportedVariables,
+               install : false)
+
+#
+# Next we make build-system variables available to the `bt` build helper Python script by injecting them into a TOML
+# file.
+#
+configure_file(input  : 'packaging/config.toml.in',
+               output : 'config.toml',
+               configuration : exportedVariables,
+               install : false)
+
+# On Linux (and TBD Mac) we inject build system variables into the markdown text that will be used to generate the man
+# page for the application.  The bt build tool script will then turn this markdown into a man page and include it in
+# the package.  (.:TBD:. This means man pages don't get installed locally, but that could be fixed by extending bt to
+# wrap around all meson operations.)
+configure_file(input  : 'doc/manpage.1.md.in',
+               output : 'manpage.1.md',
+               configuration : exportedVariables,
+               install : false)
+
+#
+# We also want to inject build-system variables into various platform-specific packaging scripts, control files and so
+# on.
+#
+# We ultimately want the generated files inside the 'packaging' subtree of the the build directory, but that is not
+# permitted by configure_file, so, that gets done by the bt script.
```

Comment:
You can, if you use `subdir('packaging')` and then define the configure_file() inside packaging/meson.build.

Reply:
Ah, neat.  I didn't know about that.  Thanks!

Reply:
It can also be useful for organizational purposes and to avoid unbounded growth of the root `meson.build` file. :)

For example, some people prefer to subdir into the src/ directory and append to a global list of sources to keep long lists of filenames separate from top-level logic about how to use them.